### PR TITLE
{Test} Live test pipeline

### PR DESCRIPTION
**Pull request by qwordy**
**Description**  

- Support upgrading API version in pipeline.

**Testing Guide**  

**History Notes**  

[Component Name 1] BREAKING CHANGE: az command a: Make some customer-facing breaking change.  
[Component Name 2] az command b: Add some customer-facing feature.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [ ] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [ ] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

**Review by haroldrandom — APPROVED** [review]

**Review by jiasli — DISMISSED** [review]
1. Why is the title CAPITALIZED?
2. It's better to compress the history.
2. Please add some description.

File: scripts/live_test/CLITest.yml [hunk 1]
```diff
@@ -176,16 +186,35 @@ jobs:
...
         az login -u azureclitest@azuresdkteam.onmicrosoft.com -p $(AZURECLITESTPASSWORD)
         az account set -s 0b1f6471-1bf0-4dda-aec3-cb9272f09590
-        # Sequential
-        azdev test $(Target) $(USER_LIVE) --mark serial --xml-path test_results.sequential.xml --no-exitfirst -a "-n 1 --json-report --json-report-summary --json-report-file=$(Target).report.sequential.json --html=$(Target).report.sequential.html --self-contained-html --reruns 3 -s"
-        # Parallel
-        azdev test $(Target) $(USER_LIVE) --mark "not serial" --xml-path test_results.parallel.xml --no-exitfirst -a "-n $(USER_PARALLELISM) --json-report --json-report-summary --json-report-file=$(Target).report.parallel.json --html=$(Target).report.parallel.html --self-contained-html --reruns 3 -s"
+
+        # Commit changes
+        if [[ "$(USER_USERNAME)" != "" || "$(USER_TOKEN)" != "" ]]; then
+          azdev test $(Target) --no-exitfirst -a "-n $(USER_PARALLELISM)"
```
Comment (jiasli) [inline scripts/live_test/CLITest.yml:198]: Why is `Target` not capitalized like `USER_USERNAME`?

File: scripts/live_test/CLITest.yml [hunk 2]
```diff
@@ -166,8 +166,18 @@ jobs:
...
         ls
         # ssh-keygen -f ~/.ssh/id_rsa -t rsa -N ''
-        git clone -b $(USER_BRANCH) $(USER_REPO)
-        #git clone https://github.com/Azure/azure-cli-extensions.git
+        git config --global user.email "AzurePipelines"
+        git config --global user.name "AzurePipelines"
+        # Commit changes
+        if [[ "$(USER_USERNAME)" != "" || "$(USER_TOKEN)" != "" ]]; then
+          REPO="$(USER_REPO)"
+          git clone -b $(USER_BRANCH) ${REPO:0:8}$(USER_USERNAME):$(USER_TOKEN)@${REPO:8}
```
Comment (jiasli) [inline scripts/live_test/CLITest.yml:174]: Please add some comments about this magical line.

File: scripts/live_test/CLITest.yml (hunk 1 above)
Comment (qwordy) [reply scripts/live_test/CLITest.yml:198]: one is internal, one is user variable.

**Comment by yungezz** [comment]
hi @qwordy how about merge the pipeline, the PR contains huge commit history. We can continue update after first merging.

File: scripts/live_test/CLITest.yml [hunk 3]
```diff
@@ -176,16 +186,35 @@ jobs:
...
         az account set -s 0b1f6471-1bf0-4dda-aec3-cb9272f09590
-        # Sequential
-        azdev test $(Target) $(USER_LIVE) --mark serial --xml-path test_results.sequential.xml --no-exitfirst -a "-n 1 --json-report --json-report-summary --json-report-file=$(Target).report.sequential.json --html=$(Target).report.sequential.html --self-contained-html --reruns 3 -s"
-        # Parallel
-        azdev test $(Target) $(USER_LIVE) --mark "not serial" --xml-path test_results.parallel.xml --no-exitfirst -a "-n $(USER_PARALLELISM) --json-report --json-report-summary --json-report-file=$(Target).report.parallel.json --html=$(Target).report.parallel.html --self-contained-html --reruns 3 -s"
+
+        # Commit changes
+        if [[ "$(USER_USERNAME)" != "" || "$(USER_TOKEN)" != "" ]]; then
+          azdev test $(Target) --no-exitfirst -a "-n $(USER_PARALLELISM)"
+          azdev test $(Target) --live --lf --xml-path test_results.parallel.xml --no-exitfirst -a "-n $(USER_PARALLELISM) --json-report --json-report-summary --json-report-file=$(Target).report.parallel.json --html=$(Target).report.parallel.html --self-contained-html --reruns 3 -s"
```
Comment (fengzhou-msft) [inline scripts/live_test/CLITest.yml:199]: Just to point out that some ADO agent only has 2 cores and the current default USER_PARALLELISM=8 will not be run as expected.

File: scripts/live_test/CLITest.yml [hunk 4]
```diff
@@ -176,16 +186,35 @@ jobs:
...
+        fi
+
         pwd
         ls
+        # Commit changes
+        if [[ "$(USER_USERNAME)" != "" || "$(USER_TOKEN)" != "" ]]; then
+          cd azure-cli
+          git status
+          git add .
+          git commit -m "auto commit"
```
Comment (fengzhou-msft) [inline scripts/live_test/CLITest.yml:214]: The commit is for updating recording files? The commit message could be more concrete.

File: scripts/live_test/CLITest.yml (hunk 2 above)
Comment (fengzhou-msft) [reply scripts/live_test/CLITest.yml:174]: What will happen if one of USER_USERNAME and USER_TOKEN is empty?

File: scripts/live_test/CLITest.yml (hunk 4 above)
Comment (qwordy) [reply scripts/live_test/CLITest.yml:214]: Yes. Good idea.

File: scripts/live_test/CLITest.yml (hunk 3 above)
Comment (qwordy) [reply scripts/live_test/CLITest.yml:199]: 2 cores does not mean it can only run 2 processes.
https://en.wikipedia.org/wiki/Context_switch

**Comment by qwordy** [comment]
> 
> 
> hi @qwordy how about merge the pipeline, the PR contains huge commit history. We can continue update after first merging.

Yes. Very huge. I think I need to delete the branch and checkout a new branch from dev with same name. Then it is very clean.

File: scripts/live_test/CLITest.yml (hunk 3 above)
Comment (qwordy) [reply scripts/live_test/CLITest.yml:199]: Logs in pipeline
```
RUNNING TESTS LIVE
The tests are set to run against current profile "latest"
============================= test session starts ==============================
platform linux -- Python 3.8.5, pytest-6.0.2, py-1.9.0, pluggy-0.13.1 -- /home/vsts/work/1/env/bin/python
[gw0] linux Python 3.8.5 cwd: /home/vsts/work/1

[gw1] linux Python 3.8.5 cwd: /home/vsts/work/1

[gw2] linux Python 3.8.5 cwd: /home/vsts/work/1

[gw3] linux Python 3.8.5 cwd: /home/vsts/work/1

[gw4] linux Python 3.8.5 cwd: /home/vsts/work/1

[gw5] linux Python 3.8.5 cwd: /home/vsts/work/1

[gw6] linux Python 3.8.5 cwd: /home/vsts/work/1

[gw7] linux Python 3.8.5 cwd: /home/vsts/work/1

[gw0] Python 3.8.5 (default, Jul 21 2020, 12:22:34)  -- [GCC 7.5.0]

[gw1] Python 3.8.5 (default, Jul 21 2020, 12:22:34)  -- [GCC 7.5.0]

[gw2] Python 3.8.5 (default, Jul 21 2020, 12:22:34)  -- [GCC 7.5.0]

[gw3] Python 3.8.5 (default, Jul 21 2020, 12:22:34)  -- [GCC 7.5.0]

[gw4] Python 3.8.5 (default, Jul 21 2020, 12:22:34)  -- [GCC 7.5.0]

[gw5] Python 3.8.5 (default, Jul 21 2020, 12:22:34)  -- [GCC 7.5.0]

[gw6] Python 3.8.5 (default, Jul 21 2020, 12:22:34)  -- [GCC 7.5.0]

[gw7] Python 3.8.5 (default, Jul 21 2020, 12:22:34)  -- [GCC 7.5.0]
gw0 [216] / gw1 [216] / gw2 [216] / gw3 [216] / gw4 [216] / gw5 [216] / gw6 [216] / gw7 [216]
```

**Pull request merged**